src/Sylius/Bundle/ApiBundle/Controller/TerminateCatalogPromotionAction.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Controller;
+
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\TerminateCatalogPromotion;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+/** @experimental */
+final class TerminateCatalogPromotionAction
+{
+    public function __construct(
+        private MessageBusInterface $commandBus,
+        private RepositoryInterface $catalogPromotionRepository,
+    ){
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $code = $request->attributes->get('code');
+
+        /** @var CatalogPromotionInterface|null $catalogPromotion */
+        $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $code]);
+
+        if (null === $catalogPromotion) {
+            return new Response(status: Response::HTTP_NOT_FOUND);
+        }
```

Comment:
[Here](https://github.com/Sylius/Sylius/pull/14270/files#diff-6b6691add61aa0c437440437a343c2469edb4cb206231ac173bbf47149d19606R40-R44) we have the same check. Maybe we should leave the one in the command handler and catch the exception here? 🤔 It just seems as the unneeded duplication